cypress_test/integration_tests/desktop/writer/top_toolbar_spec.js

```diff
@@ -330,4 +331,73 @@ describe('Top toolbar tests.', function() {
 			.should('exist');
 	});
 
+	it('Save.', function() {
+		cy.get('#tb_editbar_item_bold')
+			.click();
+
+		cy.get('#tb_editbar_item_save')
+			.click();
+
+		helper.beforeAll(testFileName, 'writer', true);
+
+		helper.reselectAllText();
+
+		cy.get('#copy-paste-container p b')
+			.should('exist');
+
+	});
+
+	it('Print', function() {
+		// A new window should be opened with the PDF.
+		cy.window()
+			.then(function(win) {
+				cy.stub(win, 'open');
+			});
+
+		cy.get('#tb_editbar_item_print')
+		    .click();
+
+		cy.window().its('open').should('be.called');
+	});
+	
+	it('Apply Undo/Redo.', function() {
+		cy.get('#tb_editbar_item_italic')
+			.click();
+
+		helper.selectAllText(false);
+
+		cy.get('#copy-paste-container p i')
+			.should('exist');
+		helper.selectAllText(false);
```

Comment:
As I see this is not needed here. For pushing undo, we don't need text selection, right?